project/SparkBuild.scala

```diff
@@ -294,7 +294,7 @@ object SparkBuild extends PomBuild {
     javaOptions ++= {
       val versionParts = System.getProperty("java.version").split("[+.\\-]+", 3)
       var major = versionParts(0).toInt
-      if (major >= 16) Seq("--add-modules=jdk.incubator.vector") else Seq.empty
+      if (major >= 16) Seq("--add-modules=jdk.incubator.vector,jdk.incubator.foreign", "-Dforeign.restricted=warn") else Seq.empty
```

Comment:
I assume the same, that these flags are required to use the vectorized code, but, that's on someone who's experimenting with Java 16 at this point.

Reply:
Yes, to use the code relying on the Vector API, one would need to pass `--add-modules=jdk.incubator.vector`. Otherwise, it will fall back to pure JDK8-compatible code. And the `--add-modules=jdk.incubator.foreign -Dforeign.restricted=warn` is for a native-wrapper BLAS implementation based on the [Foreign Linker API](https://openjdk.java.net/jeps/389). If that's not available, it automatically falls back to the `com.github.fommil` native wrappers.